mathics/builtin/graphics.py

```diff
@@ -725,6 +841,9 @@ def distance(a, b):
                 return distance(c1, c2)
         except ColorError:
             return
+        except NotImplementedError:
+            evaluation.message('ColorDistance', 'invdist', distance_function)
+            return
```

Comment:
It's a sort of hack to handle invalid custom functions. distance(a, b) returns this exception when fed with an invalid distance function, so I thought I could catch it this way. Is there a preferred way to check if any expression is a valid (pure) function? My strategy so far is applying the expression to the values and check the result (and catching exceptions).